crates/optimism/rpc/src/eth/ext.rs

```diff
@@ -19,33 +20,48 @@ const MAX_CONDITIONAL_EXECUTION_COST: u64 = 5000;
 /// Separate from [`super::OpEthApi`] to allow to enable it conditionally,
 #[derive(Clone)]
 #[allow(dead_code)]
-pub(crate) struct OpEthApiExt<N: OpNodeCore> {
-    /// Gateway to node's core components.
-    inner: Arc<OpEthApiInner<N>>,
+#[derive(Debug)]
+pub struct OpEthExtApi<N: OpNodeCore> {
+    /// Sequencer client, configured to forward submitted transactions to sequencer of given OP
+    /// network.
+    sequencer_client: Option<SequencerClient>,
+    /// The transaction pool of the node.
+    pool: N::Pool,
+    /// The provider type used to interact with the node.
+    provider: N::Provider,
```

Comment:
cool, this is better,

can we also wrap these in an additional Arc inner type, this way the clone per request is even cheaper